### ISSUE TITLE
IDE freeze

### ISSUE THREAD

**Issue by matej21** [juzna/intellij-nette#18]
```
"AWT-EventQueue-0 9.0#PS-141.1306, eap:true" prio=0 tid=0x0 nid=0x0 runnable
     java.lang.Thread.State: RUNNABLE

	at com.jetbrains.php.lang.lexer._PhpLexer.advance(_PhpLexer.java:1102)
	at com.intellij.lexer.FlexAdapter.locateToken(FlexAdapter.java:95)
	at com.intellij.lexer.FlexAdapter.getTokenType(FlexAdapter.java:58)
	at com.intellij.lexer.MergingLexerAdapterBase.locateToken(MergingLexerAdapterBase.java:71)
	at com.intellij.lexer.MergingLexerAdapterBase.getTokenType(MergingLexerAdapterBase.java:46)
	at com.intellij.lang.impl.PsiBuilderImpl.a(PsiBuilderImpl.java:207)
	at com.intellij.lang.impl.PsiBuilderImpl.<init>(PsiBuilderImpl.java:175)
	at com.intellij.lang.impl.PsiBuilderImpl.<init>(PsiBuilderImpl.java:135)
	at com.intellij.lang.impl.PsiBuilderImpl.<init>(PsiBuilderImpl.java:183)
	at com.intellij.lang.impl.PsiBuilderFactoryImpl.createBuilder(PsiBuilderFactoryImpl.java:52)
	at com.intellij.psi.tree.ILazyParseableElementType.doParseContents(ILazyParseableElementType.java:62)
	at com.intellij.psi.tree.IFileElementType.parseContents(IFileElementType.java:43)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.f(LazyParseableElement.java:172)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getLastChildNode(LazyParseableElement.java:218)
	at com.intellij.psi.impl.source.tree.LazyParseableElement.getLastChildNode(LazyParseableElement.java:36)
	at com.intellij.psi.impl.source.tree.SharedImplUtil.getLastChild(SharedImplUtil.java:57)
	at com.intellij.psi.impl.source.PsiFileImpl.getLastChild(PsiFileImpl.java:774)
	at com.intellij.psi.stubs.DefaultStubBuilder.buildStubTreeFor(DefaultStubBuilder.java:72)
	at com.intellij.psi.stubs.DefaultStubBuilder.buildStubTree(DefaultStubBuilder.java:36)
	at com.intellij.psi.stubs.StubTreeBuilder.buildStubTree(StubTreeBuilder.java:102)
	at com.intellij.psi.stubs.StubTreeLoaderImpl.readOrBuild(StubTreeLoaderImpl.java:59)
	at com.intellij.psi.impl.source.PsiFileImpl.getStubTree(PsiFileImpl.java:681)
	at com.intellij.psi.stubs.StubProcessingHelperBase.processStubsInFile(StubProcessingHelperBase.java:58)
	at com.intellij.psi.stubs.StubProcessingHelperBase.processStubsInFile(StubProcessingHelperBase.java:39)
	at com.intellij.psi.stubs.StubIndexImpl$2.perform(StubIndexImpl.java:274)
	at com.intellij.psi.stubs.StubIndexImpl$2.perform(StubIndexImpl.java:265)
	at com.intellij.util.indexing.ValueContainer.forEach(ValueContainer.java:81)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:265)
	at com.intellij.psi.stubs.StubIndexImpl.processElements(StubIndexImpl.java:238)
	at com.intellij.psi.stubs.StubIndex.process(StubIndex.java:76)
	at com.intellij.psi.stubs.StubIndex.process(StubIndex.java:95)
	at com.intellij.psi.stubs.StubIndexImpl.get(StubIndexImpl.java:227)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:144)
	at com.intellij.psi.stubs.StubIndex.getElements(StubIndex.java:134)
	at com.jetbrains.php.PhpIndexImpl.getClassesByNameInScope(PhpIndexImpl.java:217)
	at com.jetbrains.php.PhpIndexImpl.getClassesByName(PhpIndexImpl.java:211)
	at com.jetbrains.php.PhpIndexImpl.getClassesByFQN(PhpIndexImpl.java:472)
	at com.jetbrains.php.lang.psi.resolve.types.PhpType.findSuper(PhpType.java:357)
	at com.jetbrains.php.lang.psi.resolve.types.PhpType.findSuper(PhpType.java:341)
	at com.jetbrains.php.lang.psi.resolve.types.PhpType.isConvertibleFrom(PhpType.java:295)
	at cz.juzna.intellij.nette.utils.MagicFieldsUtil.findMagicFields(MagicFieldsUtil.java:50)
	at cz.juzna.intellij.nette.typeProvider.NetteObjectFieldsTypeProvider.getType(NetteObjectFieldsTypeProvider.java:43)
	at com.jetbrains.php.lang.psi.resolve.types.PhpTypeInfo$1.internalCompute(PhpTypeInfo.java:43)
	at com.jetbrains.php.lang.psi.resolve.types.PhpTypeInfo$1.internalCompute(PhpTypeInfo.java:37)
	at com.intellij.openapi.util.Computable$NotNullCachedComputable.compute(Computable.java:52)
.....

```

**Comment by Cactucs** [comment]
Freezes all the time... 
https://gist.github.com/Cactucs/d9af829687b242d64c62

**Comment by matej21** [comment]
fixed by 6900bed and f7fd06d (hopefully)
